lib/ripper-tags.rb

```diff
@@ -122,6 +122,9 @@ def self.process_args(argv, run = method(:run))
       elsif !options.recursive then abort(optparse.banner)
       end
       options.format ||= File.basename(options.tag_file_name) == "TAGS" ? "emacs" : "vim"
+      if options.format == 'emacs' and options.tag_file_name == default_options.tag_file_name
```

Comment:
Looks good! The only problem is that if `-e -f ./tags` were passed by command-line arguments, even though an explicit tag filename was specified, it would still get reset to `TAGS` by above logic. That would be an edge-case bug, since explicit user input should never be overriden by defaults.

We would need to be able to differentiate if an explicit `tag_file_name` was provided by user, which we can't do now in current code (because we didn't need to before). You might need to change `tag_file_name` in `default_options` to `nil` so that we know when the option was actually passed vs. when a default should kick in.